src/modules/twap/state/emulatedPartOrdersAtom.ts

```diff
@@ -0,0 +1,55 @@
+import { atom } from 'jotai'
+
+import { OrderClass, OrderStatus, SigningScheme } from '@cowprotocol/cow-sdk'
+
+import { OrderStatus as OrderStatusInApp } from 'legacy/state/orders/actions'
+
+import { OrderWithComposableCowInfo } from 'common/types'
+
+import { twapOrdersListAtom } from './twapOrdersListAtom'
+import { twapPartOrdersListAtom } from './twapPartOrdersAtom'
+
+import { TwapOrderItem, TwapOrderStatus } from '../types'
+
+export const emulatedPartOrdersAtom = atom<OrderWithComposableCowInfo[]>((get) => {
+  const twapOrders = get(twapOrdersListAtom)
+  const twapParticleOrders = get(twapPartOrdersListAtom)
+
+  return twapParticleOrders.map<OrderWithComposableCowInfo>((order) => {
+    const parent = twapOrders[order.twapOrderId]
+
+    const creationDate = new Date(1 + (order.order.validTo - parent.order.t) * 1000)
+
+    return {
+      order: {
+        ...order.order,
+        creationDate: creationDate.toISOString(),
+        class: OrderClass.LIMIT,
```

Comment:
We might want to make a new class for TWAP, right?

Reply:
I'm not sure. `OrderClass` is an entity from the backend, they don't have a value for `TWAP`. So, to keep transparency between the backend and the frontend, I would keep `class` as `Limit` and would distinguish the type of an order by other evidence.